azurerm/internal/services/machinelearning/tests/resource_arm_machine_learning_workspace_test.go

```diff
@@ -0,0 +1,367 @@
+			{
+				Config: testAccAzureRMMachineLearningWorkspace_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "friendly_name", "test-workspace"),
+					resource.TestCheckResourceAttr(data.ResourceName, "description", "Test machine learning workspace"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "1"),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMMachineLearningWorkspace_completeUpdated(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "friendly_name", "test-workspace-updated"),
+					resource.TestCheckResourceAttr(data.ResourceName, "description", "Test machine learning workspace update"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "2"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMMachineLearningWorkspaceExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Machine Learning Workspace not found: %s", resourceName)
+		}
+
+		id, err := parse.WorkspaceID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).MachineLearning.WorkspacesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Machine Learning Workspace %q (Resource Group %q) does not exist", id.Name, id.ResourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on machinelearningservices.WorkspacesClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMMachineLearningWorkspaceDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).MachineLearning.WorkspacesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_machine_learning_workspace" {
+			continue
+		}
+
+		id, err := parse.WorkspaceID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on machinelearningservices.WorkspacesClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMMachineLearningWorkspace_basic(data acceptance.TestData) string {
+	template := testAccAzureRMMachineLearningWorkspace_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_machine_learning_workspace" "test" {
+  name                    = "acctestworkspace-%d"
```

Comment:
could we include capitals in every property that allows them:
```suggestion
  name                    = "acctest-MLW-%d"
```

Reply:
Done